cmd/server/app/server.go

```diff
@@ -321,6 +322,10 @@ func (p *Proxy) runAgentServer(o *options.ProxyRunOptions, server *server.ProxyS
 	agentServerOptions := []grpc.ServerOption{
 		grpc.Creds(credentials.NewTLS(tlsConfig)),
 		grpc.KeepaliveParams(keepalive.ServerParameters{Time: o.KeepaliveTime}),
+		grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
```

Comment:
I realize that we don't tend to keep frontend connections open as long, but that doesn't mean it doesn't happen. Should have a similar fix as https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/cmd/server/app/server.go#L188?

Reply:
From tallclair "I don't think PermitWithoutStream makes sense as long as it's a single use tunnel, so I'm inclined to just keep the defaults..." also that the current client frontend does not set a client keep alive, so it should not be needed (yet).